tests/phpunit/tests/functions.php

```diff
@@ -1096,6 +1106,28 @@ public function test_wp_extract_urls() {
 		$this->assertSame( array_slice( $original_urls, 0, 8 ), $urls );
 	}
 
+	/**
+	 * Tests for backward compatibility of `wp_extract_urls` to remove unused semicolons.
+	 *
+	 * @ticket 30580
+	 *
+	 * @covers ::wp_extract_urls
+	 */
+	public function test_wp_extract_urls_remove_semicolon() {
+		$expected = array(
+			'http://typo.com',
+			'http://example.org/?post_type=post&p=8',
+		);
+		$actual   = wp_extract_urls(
+			'
+				http://typo.com;
+				http://example.org/?post_type=;p;o;s;t;&amp;p=8;
```

Comment:
Removing the semi-colons from this seems incorrect, semi colons are valid URL characters. For a proof of concept, set the permalink structure of your WP development environment to `/%year%;%monthnum%;%post_id%/`

Reply:
I believe that semicolons were previously removed from URLs [as mentioned in this thread](https://github.com/WordPress/wordpress-develop/pull/2283#discussion_r821096013).

By initially allowing semicolons in order to detect and decode HTML entities, then stripping extraneous semicolons, this maintains backwards compatibility while supporting HTML entities like `&amp;`.


My concern is that allowing any semicolons, other than those contained in the variants of ampersand, is a BC break that is outside the scope of this ticket. Given this, the topic of allowing semicolons in the URL detection may benefit from having its own ticket and discussion.

If the proposal to allow semicolons is supported, the required changes are to remove the `str_replace()` call and instead use:

```php
$post_links = array_unique(
    array_map( 'html_entity_decode', $post_links[2] )
);
```

Reply:
+1 on opening a separate ticket to address allowing individual semicolons in returned URLs. Some headway here is still headway, right? 😂

Reply:
@ironprogrammer @costdev Agreed. I didn't realise it was an existing bug but it most certainly is.

```
wp> wp_extract_urls( 'http://example.com/p;o;s;t' );
=> phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/shell-command/src/WP_CLI/Shell/REPL.php:52:
array(1) {
  [0] =>
  string(20) "http://example.com/p"
}
```